### ISSUE TITLE
Improve Add Feed behavior

### ISSUE THREAD

**Issue by boonebgorges** [PressForward/pressforward#1185]
The current Add Feeds flow is clunky:

![Screenshot_2024-08-21_11-23-02](https://github.com/user-attachments/assets/eb0d6e0c-c09d-4181-b8b6-7b4984feaf31)

Some thoughts and ideas for improvement:

- Let's break OPML import onto its own page. It's very much an advanced-user option, and it clutters the interface here.
- We need better success messages for adding feeds. Perhaps this could include a link to manage the feed (Dashboard > PressForward > Subscribed Feeds > Edit Feed)
- We should consider doing some feed sniffing, so that if you enter the URL of (say) a site that looks like WordPress, then we convert that to the feed, or perhaps give you options as to what you mean.
- Similarly, we should add some validation, perhaps via an interstitial. This will avoid problems where someone tries to add the URL of a feed that is broken or non-existent.

**Comment by boonebgorges** [comment]
I've started by doing a bunch of work on OPML import. I moved everything to its own tab, cleaned up the markup, got rid of the media inserter for this part of the OPML upload, and improved validation and error messages. Screenshot:

![Screenshot_2024-10-25_15-35-42](https://github.com/user-attachments/assets/d1fab3e3-1f21-4ad0-8c4e-99fac5247672)

It's still not great, but this is likely such a little-used part of the platform that it's not worth sinking lots more time into it.

**Comment by boonebgorges** [comment]
I've spent a bit of time changing the way the Add Feed mechanism works. The idea was to add a bit of validation and feedback. Here's me playing with it:

![Peek 2024-10-28 15-18](https://github.com/user-attachments/assets/2b015ec3-837a-4af4-b5bd-a7b9bc4a33ba)

* When you first load the page, you see only a Check button. (This applies only to the JS version. Without JS you just see a 'Submit' button.)
* 'Check' sends an AJAX request to the server. The server tries to see whether this looks like a real URL. If you provide something without a scheme (`http://` etc), it'll add it for you.
* If the URL is valid-looking, WP tries to fetch the feed. If it's a valid feed URL, you'll get a success message.
* If the feed URL is invalid, WP also tries to find related feed URLs. For example, most WordPress-powered sites `https://example.com` have a feed URL at `https://example.com/feed`. If WP detects one of these related valid feed URLs, it'll send it back instead. I then show a message explaining that a related feed URL is found, and provide a button that lets you use it instead.
* I changed the submit button to 'Subscribe to Feed' and I only show it after validation.

Not perfect, but I think it's improved for most cases.
